executor/joiner.go

```diff
@@ -102,18 +101,25 @@ func newJoiner(ctx sessionctx.Context, joinType plan.JoinType,
 	}
 	switch joinType {
 	case plan.SemiJoin:
+		base.shallowRow = chunk.MutRowFromTypes(colTypes)
```

Comment:
What not change it on line 94 directly? So we don't repeat it multiple times?

Reply:
only `SemiJoin, AntiSemiJoin, LeftOuterSemiJoin, AntiLeftOuterSemiJoin` need `shallowRow`.